calcom/cal.com
# feat: autolink org on provider signup

## What does this PR do?

When someone uses an external provider to sign-up, such as Google, and the `ORGANIZATIONS_AUTOLINK` env variable is used, then the signed up user will be autolinked to the organization automatically as a member. 

For example, if an organization "Acme" with domain `acme.com` is created, and the organization is verified, any user signing up with Google using a `acme.com` domain such as `john@acme.com`, they will automatically be linked to organization "Acme" as a member.

 Loom Video: Coming soon

## Type of change


## How should this be tested?

> **Note**
> You don't need to have organizations enabled for these steps, only env var `ORGANIZATIONS_AUTOLINK=1` 

1. Make sure you have organizations feature turned on by logging in as Admin and going to Features section in Settings
2. Configure Google integration using the steps [here](https://github.com/calcom/cal.com#obtaining-the-google-api-credentials)
3. Change env var `NEXT_PUBLIC_WEBSITE_URL=http://localhost:3001`
4. Run `yarn dev:website`
5. Create an org ([http://localhost:3000/settings/organizations/new](http://localhost:3000/settings/organizations/new)) with an email that matches the email address used in Google signup, for example, create an org using email john.doe@cal.com if you can signup using a cal.com Google account, or for testing purposes only, create an org using email john.doe@gmail.com if you have a gmail account to then signup
6. Go to [http://localhost:3001/signup](http://localhost:3001/signup) and input a username (which will not be used after all) and click the Google icon to sign up using a Google Account
